Ghost_in_the_Shell/client_folder_react_app/ethisim/src/components/ScenarioCard.js

```diff
@@ -40,11 +40,167 @@ const useStyles = makeStyles((theme) => ({
 
 export default function ScenarioCard(props) {
   const classes = useStyles();
-  const data = props.data;
-  
+  const { data } = props;
+  const { id, scenarioName, className, finished } = data;
+
+  //If scenario is unfinished, we show the buttons "Edit," "Delete," "Share"
+  //If scenario is finished, we show the button "Edit," "Delete," "Share," "View Student Data"
+  const buttons = 
+    finished ? 
+      <Grid
```

Comment:
Very longwinded solution to figuring out which buttons to show. If anyone has any ways to make this implementation cleaner, would be greatly appreciated.

Reply:
Why not just use the ternary for the bottom row? Thats all thats changing. Would reduce a lot of the copy paste

Reply:
Got it.